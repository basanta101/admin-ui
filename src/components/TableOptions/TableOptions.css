.btn {
    padding: 2px
}

.delBtn{
    margin-right: 5px;
}