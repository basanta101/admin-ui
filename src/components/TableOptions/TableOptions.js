import classNames from "classnames";
import "./TableOptions.css";

const CellInput = ({
  handleRowDelBtnClick = (f) => f,
  handleRowEditBtnClick = (f) => f,
  onDone = (f) => f,
  isRowInEditMode = false,
}) => {
  return (
    <>
      <button
        className={classNames("delBtn", "btn")}
        onClick={handleRowDelBtnClick}
        data-testid='delBtn'
      >
        X
      </button>
      <button
        className="btn"
        onClick={() => {
          isRowInEditMode ? onDone() : handleRowEditBtnClick();
        }}
        data-testid='done_edit_Btn'
      >
        {isRowInEditMode ? "DONE" : "EDIT"}
      </button>
    </>
  );
};

export default CellInput;
