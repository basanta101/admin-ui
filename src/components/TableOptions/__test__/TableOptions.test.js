import renderer from "react-test-renderer";
import { render, fireEvent, getByTestId } from "@testing-library/react";
import TableOptions from "../TableOptions";

const setup = ({ isRowInEditMode = true } = {}) => {
  const handleRowDelBtnClick = jest.fn();
  const handleRowEditBtnClick = jest.fn();
  const onDone = jest.fn();
  const { container } = render(
    <TableOptions
      handleRowDelBtnClick={handleRowDelBtnClick}
      handleRowEditBtnClick={handleRowEditBtnClick}
      onDone={onDone}
      isRowInEditMode={isRowInEditMode}
    />
  );

  return {
    container,
    handleRowDelBtnClick,
    handleRowEditBtnClick,
    onDone,
  };
};

test("on clicking the X btn it calls the handleRowDelBtnClick", () => {
  const { container, handleRowDelBtnClick } = setup();
  fireEvent.click(getByTestId(container, 'delBtn'));
  expect(handleRowDelBtnClick).toHaveBeenCalledTimes(1);
});

test("on clicking the Edit btn it calls the handleRowEditBtnClick, when isRowInEditMode is false", () => {
  const { container, handleRowEditBtnClick } = setup({ isRowInEditMode: false});
  fireEvent.click(getByTestId(container, 'done_edit_Btn'));
  expect(handleRowEditBtnClick).toHaveBeenCalledTimes(1);
});

test("on clicking the Done btn it calls the onDone, when isRowInEditMode is true", () => {
  const { container, onDone } = setup({ isRowInEditMode: true});
  fireEvent.click(getByTestId(container, 'done_edit_Btn'));
  expect(onDone).toHaveBeenCalledTimes(1);
});

it("renders a snapshot of TableOptions Component", () => {
  const tree = renderer.create(<TableOptions />).toJSON();
  expect(tree).toMatchSnapshot();
});
