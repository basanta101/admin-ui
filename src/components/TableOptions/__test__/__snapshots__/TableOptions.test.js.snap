// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot of TableOptions Component 1`] = `
Array [
  <button
    className="delBtn btn"
    data-testid="delBtn"
    onClick={[Function]}
  >
    X
  </button>,
  <button
    className="btn"
    data-testid="done_edit_Btn"
    onClick={[Function]}
  >
    EDIT
  </button>,
]
`;
